[CI] ClassificationIT.testTooLowConfiguredMemoryStillStarts & RunDataFrameAnalyticsIT.testModelMemoryLimitLowerThanEstimatedMemoryUsage

<!--
Please fill out the following information, and ensure you have attempted
to reproduce locally
-->

**Build scan**:
https://gradle-enterprise.elastic.co/s/lqjgfg43cew4q
Note it has two failed tests

**Repro line**:
> ./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTest' --tests "org.elasticsearch.xpack.ml.integration.ClassificationIT.testTooLowConfiguredMemoryStillStarts" -Dtests.seed=B6982EE4D20C808E -Dtests.security.manager=true -Dtests.locale=es-CL -Dtests.timezone=Australia/West -Druntime.java=11
>
> ./gradlew ':x-pack:plugin:ml:qa:native-multi-node-tests:integTest' --tests "org.elasticsearch.xpack.ml.integration.RunDataFrameAnalyticsIT.testModelMemoryLimitLowerThanEstimatedMemoryUsage" -Dtests.seed=B6982EE4D20C808E -Dtests.security.manager=true -Dtests.locale=sr-RS -Dtests.timezone=America/Sitka -Druntime.java=11
>


**Reproduces locally?**:
No

**Applicable branches**:
master

**Failure history**:
<!--
Link to build stats and possible indication of when this started failing and how often it fails
<https://build-stats.elastic.co/app/kibana>
-->
The method `testModelMemoryLimitLowerThanEstimatedMemoryUsage` failed for the very first faliure in last 30 days according to [build-stats](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:b646ed00-7efc-11e8-bf69-63c8ef516157,key:build.project,negate:!f,params:(query:elasticsearch,type:phrase),type:phrase,value:elasticsearch),query:(match:(build.project:(query:elasticsearch,type:phrase)))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:b646ed00-7efc-11e8-bf69-63c8ef516157,key:build.branch,negate:!t,params:!(pull-request,pull-request-1,pull-request-2,pull-request-bwc),type:phrases,value:'pull-request,%20pull-request-1,%20pull-request-2,%20pull-request-bwc'),query:(bool:(minimum_should_match:1,should:!((match_phrase:(build.branch:pull-request)),(match_phrase:(build.branch:pull-request-1)),(match_phrase:(build.branch:pull-request-2)),(match_phrase:(build.branch:pull-request-bwc))))))),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:RunDataFrameAnalyticsIT),sort:!(process.time-start,desc))). I was gonna let it pass, but then I noticed that `testTooLowConfiguredMemoryStillStarts` is a new method added in today's [changes](https://github.com/elastic/elasticsearch/commit/44d667b33faf28fdc6a8be0e6b617403d9dc1855). Both methods are touched in the change set. So there might be something going on.

**Failure excerpt**:
```
java.lang.AssertionError: |  
-- | --
  | Expected: <0L> |  
  | but: was <1L> |  


at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) |  
-- | --
  |   | •••
  |   | at org.elasticsearch.xpack.ml.integration.MlNativeDataFrameAnalyticsIntegTestCase.cleanUpAnalytics(MlNativeDataFrameAnalyticsIntegTestCase.java:96) |  
  |   | at org.elasticsearch.xpack.ml.integration.MlNativeDataFrameAnalyticsIntegTestCase.cleanUpResources(MlNativeDataFrameAnalyticsIntegTestCase.java:87) |  
  |   | at org.elasticsearch.xpack.ml.integration.MlNativeIntegTestCase.cleanUp(MlNativeIntegTestCase.java:181) |  
  |   | at org.elasticsearch.xpack.ml.integration.RunDataFrameAnalyticsIT.cleanup(RunDataFrameAnalyticsIT.java:62)


```

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java
@@ -790,6 +790,7 @@ public class ClassificationIT extends MlNativeDataFrameAnalyticsIntegTestCase {
         putAnalytics(config);
         // Shouldn't throw
         startAnalytics(jobId);
+        waitUntilAnalyticsIsFailed(jobId);
         // It could be marked as failed...
         forceStopAnalytics(jobId);
         waitUntilAnalyticsIsStopped(jobId);

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/MlNativeDataFrameAnalyticsIntegTestCase.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/MlNativeDataFrameAnalyticsIntegTestCase.java
@@ -157,6 +157,10 @@ abstract class MlNativeDataFrameAnalyticsIntegTestCase extends MlNativeIntegTest
         assertBusy(() -> assertIsStopped(id), waitTime.getMillis(), TimeUnit.MILLISECONDS);
     }
 
+    protected void waitUntilAnalyticsIsFailed(String id) throws Exception {
+        assertBusy(() -> assertIsFailed(id), TimeValue.timeValueSeconds(30).millis(), TimeUnit.MILLISECONDS);
+    }
+
     protected List<DataFrameAnalyticsConfig> getAnalytics(String id) {
         GetDataFrameAnalyticsAction.Request request = new GetDataFrameAnalyticsAction.Request(id);
         return client().execute(GetDataFrameAnalyticsAction.INSTANCE, request).actionGet().getResources().results();
@@ -205,6 +209,11 @@ abstract class MlNativeDataFrameAnalyticsIntegTestCase extends MlNativeIntegTest
         assertThat(stats.getId(), equalTo(id));
         assertThat(stats.getFailureReason(), is(nullValue()));
         assertThat("Stats were: " + Strings.toString(stats), stats.getState(), equalTo(DataFrameAnalyticsState.STOPPED));
+    }
+
+    protected void assertIsFailed(String id) {
+        GetDataFrameAnalyticsStatsAction.Response.Stats stats = getAnalyticsStats(id);
+        assertThat("Stats were: " + Strings.toString(stats), stats.getState(), equalTo(DataFrameAnalyticsState.FAILED));
     }
 
     protected void assertProgressIsZero(String id) {

--- a/x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/RunDataFrameAnalyticsIT.java
+++ b/x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/RunDataFrameAnalyticsIT.java
@@ -519,7 +519,7 @@ public class RunDataFrameAnalyticsIT extends MlNativeDataFrameAnalyticsIntegTest
         assertIsStopped(id);
         //should not throw
         startAnalytics(id);
-        // Might have been marked as failed
+        waitUntilAnalyticsIsFailed(id);
         forceStopAnalytics(id);
         waitUntilAnalyticsIsStopped(id);
     }

